### DISCUSSION TITLE
Using the assistants and threads APIs to prompt GPT by adding files

### DISCUSSION THREAD

**Discussion by Duncanr-glitch** [microsoft/TypeChat#151] [Q&A]
Is there currently a way to use TypeChat to prompt GPT to be able to use the assistants and threads APIs? The main thing I want to be able to do is upload files to have GPT have that context when answering my questions.

https://platform.openai.com/docs/assistants/overview

**Comment by DanielRosenwasser** [comment]
I'm not entirely familiar with assistants and threads yet, but you can often tailor anything for TypeChat by creating a distinct [`TypeChatLanguageModel`](https://github.com/microsoft/TypeChat/blob/0936d84e3149c64fd5323fdfabe36f27ec1e763d/src/model.ts#L26C18-L41) and threading it into whatever translator you have. If I get the chance to dive deeper, I might have a better answer later (or others should feel free to chime in!)
